### ISSUE TITLE
Babel ignoring `only`

### ISSUE THREAD

**Issue by jwalton** [babel/babel#5860]
So, I have a project X, which depends on a project Y:

    /Users/jwalton/benbria/X
    /Users/jwalton/benbria/Y

The files inside X/src get compiled to X/lib, and ditto for Y/src and Y/lib.

Inside X, I `npm link @benbria/Y`.  When I try to run Babel, I would frequently run into problems where Babel would try to process files from /Users/jwalton/benbria/Y/**/*.js - I think this happens because they are not being loaded out of the X/node_modules folder anymore, by virtue of the `link`.  No problem, I add an `only` to my config:

```js
{
    presets: [ 'react', 'es2015', 'es2016' ],
    only: [
        '/Users/jwalton/benbria/X/src/*',
        '/Users/jwalton/benbria/X/src/**/*'
     ],
     sourceMaps: false
}
```

And all is well... Until very recently, when I started running into this again, even though I haven't changed anything, for my babelify files:

```
[17:16:54] Error compiling 'managerInbox'. { Error: Couldn't find preset "es2016" relative to directory "/Users/jwalton/benbria/loop-common/lib/utils" while parsing file: /Users/jwalton/benbria/loop-common/lib/utils/escapeHtml.js
```

So, Y doesn't have the `es2016` plugin installed, and Babel is searching for the plugin from the compiled file (instead of from the run directory, which is weird to begin with), and not finding it... 
 Except, Babel should not be trying to compile `/Users/jwalton/benbria/Y/lib/utils/escapeHtml.js` in the first place.  I tried adding a `console.log()` [here](https://github.com/babel/babelify/blob/6d45fa84762f3b57a4b7bef0e1bbdf484614f74d/index.js#L64), and sure enough my config has the `only` in it (in fact, it's the config I copy pasted above).  

Using babel-core 6.25.0, and babelify 7.3.0.

**Comment by basedjux** [comment]
I'm going under the assumption that the files you're wanting to include are all .js files...
Have you tried this:

```js
{
    presets: [ 'react', 'es2015', 'es2016' ],
    only: [
        '/Users/jwalton/benbria/X/src/*.js',
        '/Users/jwalton/benbria/X/src/**/*.js'
     ],
     sourceMaps: false
}
```

I say this because I noticed in 

/packages/babel-core/api.js#630-633

it has:

```js
      transformAsync("", {
        only: ["/foo/node_modules/*.bar"],
        filename: "/foo/node_modules/foo.bar",
      }).then(assertNotIgnored),
```

Just a thought. Could be globbing is getting mangled somehow.

**Comment by hzoo** [comment]
`only` has had some issues, should be fixed in Babel 7 alpha releases (7.0.0-alpha.12). Your probably going to have problems with npm linking and finding presets/plugins as well in 6.x that are being worked on it 7.
